src/allmydata/test/web/test_web.py

```diff
@@ -127,6 +127,7 @@ def __init__(self, p, history):
         import mock
         self.progress = mock.Mock()
         self.progress.progress = 100.0
+        self.size = 1234
```

Comment:
I wonder if this should just be a `QueuedItem`... It's a pure in-memory structure, it doesn't seem like it should hurt to use it in the tests.  And doing so avoids the need to duplicate its structure like this.

Reply:
Probably not really actionable for this PR but this seems like the kind of functionality that would be well and easily tested using Hypothesis stateful tests.

Reply:
I made it a plain `QueuedItem` and that works fine.